scripting/lua/api/netpacks/BattleUnitsChanged.cpp

```diff
@@ -44,7 +44,7 @@ int BattleUnitsChangedProxy::add(lua_State * L)
 	if(!S.tryGet(1, object))
 		return S.retVoid();
 
-	uint32_t id;
+	uint32_t id = 0;
 
 	if(!S.tryGet(2, id))
 		return S.retVoid();
```

Comment:
I think this is superfluous. If I read the logic right: if `id` can't be filled from Lua, return immediately.

Reply:
@kambala-decapitator this is filled because clang-tidy initialized all variables. I can block this with //NOLINT, but do I need to block it?

Reply:
Then don't use it? Or significantly clean up its output?
So far I don't see any significant improvements neither here nor in #1559

Reply:
@IvanSavenko some constructors and destructors made empty by clang-tidy itself and got defaulted by next passes. Human would not detect it as is, if he will not review a code in details.

Reply:
Example? Most complex case I saw is `ObstacleSideOptions::ObstacleSideOptions` [here](https://github.com/vcmi/vcmi/pull/1559/files#diff-b371b0903dfc2aa65381f853eef4f160dd6e624a48edc4ca095550124bcc9d27).
Still clearly an empty constructor that only has default-init of member and parent class.

Nice, but not something worth spending time on.

Reply:
@IvanSavenko here is an example of defaulting a huge copy constructor from https://github.com/vcmi/vcmi/pull/1554/commits/0b22bf0f3f51ee0e102c2effaddb21ffe1b96c88#diff-7906420fb6fbc1c2a5f88c4dc055c77fa91e77aadf5b5f56dda376e2e85a2323

This is from first attempt of clang-tidy, but second attempt will do the same.

Reply:
Nice, although seems to be the only case, and poorly written even from c++03 perspective.
Would be useful later, for c++20 default comparisons though.

Reply:
@IvanSavenko I can remove checks you do not like, like init-all-variables and do modernization again)

Reply:
@IvanSavenko cppcoreguidelines-init-all removed